### PR TITLE
Feat: Add Support for Google Palm

### PR DESCRIPTION
**Pull request by WaseemSabir**
This PR aims to add support for Google Palm API.

- [x] Tests added and passed if fixing a bug or adding a new feature
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai#contributing).

### CODE REVIEW

File: pandasai/llm/google_palm.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from google import generativeai
+
+from ..exceptions import APIKeyNotFoundError
+from .base import LLM
+
+
+class GooglePalm(LLM):
+    """Google Palm LLM"""
+
+    model: str = "models/text-bison-001"
```
Comment (gventuri) [inline pandasai/llm/google_palm.py:13]: Can we make it so we support multiple models (in a similar way we did with OpenAI)

**Review by gventuri — CHANGES_REQUESTED** [review]

**Comment by gventuri** [comment]
@WaseemSabir great job, just a minor comment!

P.s. check the tests, seems something is failing!

File: tests/llms/test_google_palm.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            ValueError, match=re.escape("top_p must be in the range [0.0, 1.0]")
+        ):
+            GooglePalm(api_key="test", top_p=-1).call("Hello", "World")
+            GooglePalm(api_key="test", top_p=1.1).call("Hello", "World")
+
+        with pytest.raises(
+            ValueError, match=re.escape("top_k must be in the range [0.0, 1.0]")
+        ):
+            GooglePalm(api_key="test", top_k=-1).call("Hello", "World")
+            GooglePalm(api_key="test", top_k=1.1).call("Hello", "World")
```
Comment (mspronesti) [inline tests/llms/test_google_palm.py:59]: Beware that these only test that **at least one** of the statements under `with pytest.raises()` raises the specified exception with the mentioned message so if your purpose was to test both the limits of `temperature`, `top_p` and `top_k` I fear here you're not doing it properly.

Comment (WaseemSabir) [reply tests/llms/test_google_palm.py]: Thanks for highlighting it! I've fixed it.

File: pandasai/llm/google_palm.py (hunk 1 above)
Comment (WaseemSabir) [reply pandasai/llm/google_palm.py:8]: @gventuri I have refactored it. Please check again.

**Pull request merged**